needed for build tools master-2021-02-19T21-48-54; Update helm linting issue; requires k8s 1.16

jacob-delgado opened the pull request:
See discussion at https://github.com/istio/tools/pull/1450. This is done to prepare master for the new build tools image.

As discussed previously in T&R, we no longer need to support k8s 1.15. Helm linting in 3.5.2 marks mutating webhook and validating webhook configuration as a warning due to v1beta1 usage. Update to v1, as it is supported in k8s 1.16+

Update the dockerfile for centos to ignore hadolint errors. These are similar to the ones already ignored for debian packages.


[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure


Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any changes that may affect Istio users.

Commit:
Update helm linting issue; requires k8s 1.16

Commit:
Ignore hadolint issues

* DL3031: Do not use yum update
* DL3032: yum clean all missing after yum command.
* DL3033: Specify version with yum install -y <package>-<version>

istio-testing commented:
@jacob-delgado: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio | 5dce85f5152ee228c5bf6e82461bc39c4df71e40 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/30970/gencheck_istio/1362911404453007360) | `/test gencheck_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@jacob-delgado: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_istio | 5dce85f5152ee228c5bf6e82461bc39c4df71e40 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/30970/gencheck_istio/1362911404453007360) | `/test gencheck_istio`
integ-helm-tests_istio | 5dce85f5152ee228c5bf6e82461bc39c4df71e40 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/30970/integ-helm-tests_istio/1362911401923842048) | `/test integ-helm-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

howardjohn reviewed (commented):
I see a lot more usages of admissionregistration.k8s.io/v1beta1 than just these 3 files. Maybe need to run `make gen`

howardjohn commented:
Does it not complain about the deprecated CRD? since that fix is much more involved

istio-testing commented:
@jacob-delgado: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-helm-tests_istio | 060ee3683a4f3ed3dcc5349e2d81c528834fbe54 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/30970/integ-helm-tests_istio/1362915953486073856) | `/test integ-helm-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

Commit:
Remove references to v1beta1

Update gen-istio.yaml
Update gen-istiod-remote.yaml
run make gen
Remove v1beta1 admissionsregistration
Package up gz

istio-testing commented:
@jacob-delgado: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-helm-tests_istio | 73f9dc64ae67447bfda3338fe4c8e742ffda9193 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/30970/integ-helm-tests_istio/1362959212774166528) | `/test integ-helm-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

jacob-delgado commented:
> I see a lot more usages of admissionregistration.k8s.io/v1beta1 than just these 3 files. Maybe need to run `make gen`

There's a tar.gz file that contains the manifest. Figured it out.

> Does it not complain about the deprecated CRD? since that fix is much more involved

Given that I'm going to close this and use the version of helm we have as a dependency. When I also tried to update the helm library to 3.5.2 I was getting go mod issues. We can do that at a different tie.

Closing in lieu of https://github.com/istio/tools/pull/1451

I'll open another PR with just the hadolint fixes
